engineering/frontend.md

```diff
@@ -14,6 +14,14 @@ There are tons of opinion around project structure, each with it's own pros and
 - Styling: [TailwindCSS](https://tailwindcss.com/docs/what-is-tailwind/) - a ultility-based CSS library to make writing CSS in React (an other client-side UI libraries/frameworks) fast and enjoyable. We also plugged in [PurgeCSS](https://github.com/dwarvesf/template-react-app/blob/master/template/purgecss.config.js) to remove unused CSS classes generated by Tailwind in production build
 - [Pre-configured ESLint](https://github.com/dwarvesf/template-react-app/blob/master/template/.eslintrc.js) with Prettier integration
 
+### Structure with Create React App
+- Global state management: [Redux](https://redux.js.org/). For GraphQL we prefer using [Apollo-client](https://www.apollographql.com/docs/react/)
+- Routing: [React-router](https://reacttraining.com/react-router/)
+- Localization: [I18n-js](https://github.com/fnando/i18n-js) or [React-intl](https://github.com/formatjs/react-intl)
+- Styling: SCSS or LESS
+- Linter: [ES-Lint](https://eslint.org/)
+Actually we prefer using typescript for type checker instead of using prop-types
```

Comment:
need @zlatanpham ideas on this, I get the idea but want more *opinionated* details. Best practices are what we _recommend_ therefor no "or" here, guide people with *your* choices, please

Reply:
Agreed. If we state we prefer to use X, we should also mention why, what problems it solved for us.

Reply:
FYI, I'm drafting my own CL too.